### ISSUE TITLE
Update card is not displayed when editing a saved card

### ISSUE THREAD

**Issue by dragosb01** [mozilla-mobile/firefox-ios#20482]
## Steps to reproduce
Follow the steps from below TC
or
1. Make sure you have a credit card already added
2. Access https://checkout.stripe.dev/preview 
3. Select a card already saved
4. Modify the card details
5. Tap Pay button
### Expected behavior
Update card prompt should be displayed
### Actual behavior
Securely save this card prompt is displayed
### Device & build information
* Device: iPhone 15 simulator iOS 17.4
* Build version: 128
* First seen version: Yes
### Notes
https://testrail.stage.mozaws.net/index.php?/cases/view/2306980
Automation test affected: XCUITests/CreditCardsTests#testUpdatePrompt
Attachments: 
<img width="380" alt="Screenshot 2024-05-30 at 10 46 55" src="https://github.com/mozilla-mobile/firefox-ios/assets/134391433/b5ca7f7e-016e-418e-93fc-1239c2e74192">

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-9247)

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Razvan Litianu can you take a look to this one?

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

Some findings:

  * When the expiration date field is updated after being autofilled, and the form is submitted, regardless of the new expiration date, the old value is read back into the client (hence why the “update” card does not display). So the problem seems to be in the content script reading back in the update field data
  * This issue seems to only occur when modifying the expiration date field after it has been autofilled (everything works fine when changing the cardholder name field after autofilling)
    * Everything also works fine if the credit card details are entered in manually

  * This issue seems to only occur when the expiration date field is a text field. Things seem to work as expected when the expiration date is controlled by dropdowns

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

The payload (WKScriptMessage) seems to contain old field values (from autofill). Issam Mani could this be an content script issue?

**Comment by data-sync-user** [comment]
➤ Issam Mani commented:

It’s tracked here: https://mozilla-hub.atlassian.net/browse/FXIOS-9026 ( https://mozilla-hub.atlassian.net/browse/FXIOS-9026|smart-link ) . And this ticket is blocked by another one from Desktop.

**Comment by data-sync-user** [comment]
➤ Issam Mani commented:

{quote}The payload (WKScriptMessage) seems to contain old field values (from autofill). Issam Mani could this be an content script issue?{quote}

I documented what happens in this comment ( https://bugzilla.mozilla.org/show_bug.cgi?id=1896494#c1 ). TL;DR we have hidden fields and we fill those and this causes the issue. This could happen for any split field not only expiration date.

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

Closing as duplicate of https://mozilla-hub.atlassian.net/browse/FXIOS-9026 ( https://mozilla-hub.atlassian.net/browse/FXIOS-9026|smart-link )

**Comment by dragosb01** [comment]
Closing duplicate task
